src/main/webapp/script.js

```diff
@@ -71,15 +68,14 @@ function postData(url, data) {
         body: data
       }
   )
-      .then((response) => {
-        if (!response.ok) {
-          throw new Error('Network error detected');
-        }
-        return response.json();
-      })
+      .then((response) => getJsonIfOk(response))
       .catch((error) => handleFetchError(error));
 }
 
-function handleFetchError(exception) {
-  console.error(exception);
+function getJsonIfOk(response) {
+  if (!response.ok) {
+    console.error('Network error detected');
```

Comment:
You can still `throw` here. You even should, as without the throw it would try to convert to json on line79.

Reply:
I actually wanted to put the return in an else. Would that be an appropriate solution?

Reply:
Even if you do that, you still have to return something here. And that would be attempted to be json deserialized...
So Jacek's suggestion to throw is actually better and you can share the error handling/logging logic in `handleFetchError`

Reply:
I think we even discussed something similar in the portfolio project to arrive at this code:
https://github.com/fuiszgt/Step2020portfolio/blob/master/portfolio/src/main/webapp/fetch.js

Reply:
Ok. I think I have done it, please check out :)